include/data/json.h

```diff
@@ -16,6 +16,15 @@
 extern "C" {
 #endif
 
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
+#define UNNAMED_UNION_BEGIN
+#define UNNAMED_UNION_END
+#else
+#define UNNAMED_UNION_BEGIN	{
+#define UNNAMED_UNION_END	},
+#endif
```

Comment:
Do we really need `UNNAMED_UNION_BEGIN` and `UNNAMED_UNION_END`?

Since adding curly braces will not cause an error on later GCC versions, I think it would be better to just use the curly braces, as was done in #19069.

Reply:
Will do.